new/default-resource-name.md

```diff
@@ -0,0 +1,103 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Default Resource Name
+
+In which resources get names and DSL by default, unless you say you don't want one.
+
+## Motivation
+
+    As a Chef user,
+    I want convention to drive my DSL,
+    So that I can concentrate on my code instead of ceremony.
+
+## Specification
+
+When a user declares a resource class thus:
+
+```ruby
+class MyResource < Chef::Resource
+end
+```
+
+It will automatically gain a resource_name and DSL derived from the class:
+
+```ruby
+my_resource 'foo' do
+end
+```
+
+This is already (currently) the case for classes declared in Chef::Resource; this extends the rule to all classes anywhere and removes the special case.
+
+### resource_name creates automatic DSL
+
+When a class's resource_name is set, it will always get DSL will that resource name. Thus, this will work:
+
+```ruby
+class X < Chef::Resource
+  resource_name :my_resource
+end
+my_resource 'foo' do
+end
+```
+
+#### New resource classes' resource names are set automatically
+
+When a resource class is declared with an explicit name, its resource name is set automatically:
+
+```ruby
+class SuperAwesome < Chef::Resource
```

Comment:
So to expand on this a bit from a discussion on IRC (this is going to be a wall of text, bear with me). My concern is that I would like to see The System encourage better layout of the resource and provider classes. Right now we generally have `Chef::Resource::Name` and `Chef::Provider::Name` which is gross because the two classes are totally separated. This is a lost cause for those namespaces themselves, but we've started to see some better (IMO) layouts in third-party code. Two notables are my layout (`Name::Resource` and `Name::Provider`) and @jkeiser's layout (`Name` and `Name::Provider`). I think we should encourage the use of layouts like that by building the name extraction code to support them. It is true you can always set the name manually so every layout is always "possible", but the default behavior sets developer expectations and we have a chance to possibly reshape those here so I think we should :-)

From the two examples I gave, possibly just taking the class name and chopping off a few known words could go a long way.